app/src/common/gecko/com/igalia/wolvic/browser/api/impl/PromptDelegateImpl.java

```diff
@@ -0,0 +1,844 @@
+        return map(mDelegate.onBeforeUnloadPrompt(mSession, new BeforeUnloadPromptImpl(prompt) {
+            @NonNull
+            @Override
+            public WSession.PromptDelegate.PromptResponse confirm(@Nullable WAllowOrDeny allowOrDeny) {
+                return new PromptResponseImpl(mGeckoPrompt.confirm(toGecko(allowOrDeny)));
+            }
+        }));
+    }
+
+    @Nullable
+    @Override
+    public GeckoResult<PromptResponse> onRepostConfirmPrompt(@NonNull GeckoSession session, @NonNull RepostConfirmPrompt prompt) {
+        return map(mDelegate.onRepostConfirmPrompt(mSession, new RepostConfirmPromptImpl(prompt) {
+            @NonNull
+            @Override
+            public WSession.PromptDelegate.PromptResponse confirm(@Nullable WAllowOrDeny allowOrDeny) {
+                return new PromptResponseImpl(mGeckoPrompt.confirm(toGecko(allowOrDeny)));
+            }
+        }));
+    }
+
+    @Nullable
+    @Override
+    public GeckoResult<PromptResponse> onButtonPrompt(@NonNull GeckoSession session, @NonNull ButtonPrompt prompt) {
+        return map(mDelegate.onButtonPrompt(mSession, new ButtonPromptImpl(prompt) {
+            @Nullable
+            @Override
+            public String message() {
+                return mGeckoPrompt.message;
+            }
+
+            @NonNull
+            @Override
+            public WSession.PromptDelegate.PromptResponse confirm(int selection) {
+                return new PromptResponseImpl(mGeckoPrompt.confirm(selection));
+            }
+        }));
+    }
+
+    @Nullable
+    @Override
+    public GeckoResult<PromptResponse> onTextPrompt(@NonNull GeckoSession session, @NonNull TextPrompt prompt) {
+        return map(mDelegate.onTextPrompt(mSession, new TextPromptImpl(prompt) {
+            @Nullable
+            @Override
+            public String message() {
+                return mGeckoPrompt.message;
+            }
+
+            @Nullable
+            @Override
+            public String defaultValue() {
+                return mGeckoPrompt.defaultValue;
+            }
+
+            @NonNull
+            @Override
+            public WSession.PromptDelegate.PromptResponse confirm(@NonNull String text) {
+                return new PromptResponseImpl(mGeckoPrompt.confirm(text));
+            }
+        }));
+    }
+
+    @Nullable
+    @Override
+    public GeckoResult<PromptResponse> onAuthPrompt(@NonNull GeckoSession session, @NonNull AuthPrompt prompt) {
+        return map(mDelegate.onAuthPrompt(mSession, new AuthPromptImpl(prompt) {
+            @Nullable
+            @Override
+            public String message() {
+                return mGeckoPrompt.message;
+            }
+
+            @NonNull
+            @Override
+            public AuthOptions authOptions() {
+                int flags = 0;
+                if ((mGeckoPrompt.authOptions.flags & AuthPrompt.AuthOptions.Flags.CROSS_ORIGIN_SUB_RESOURCE) != 0) {
```

Comment:
Do we need the `!= 0` comparison? Same for the other if conditions.

Reply:
yes, Java doesn't convert int to boolean